### PR TITLE
Added functionality for dumping the codex out as a website.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Adds a verb and handlers for writing the entire codex out as HTML at runtime.

Opening this so I don't forget it exists. Still needs a solution to the searchability problem.

## Why and what will this PR improve
Self-generating wiki.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Search isn't implemented but the rest is working. Calling this done for now.

File: code/modules/codex/entries/_codex_entry.dm [hunk 1]
```diff
@@ -63,10 +63,10 @@
 	if(!name)
 		if(length(associated_strings))
 			name = associated_strings[1]
-		else
+		else if(store_codex_entry)
 			CRASH("Attempted to instantiate unnamed codex entry with no associated strings!")
 
-	if(store_codex_entry)
+	if(name && store_codex_entry)
```
Comment (out-of-phaze) [inline code/modules/codex/entries/_codex_entry.dm:76]: I feel like this can be simplified or changed somehow, but it should work fine as-is

**Review by out-of-phaze — APPROVED** [review]
aside from searchability, this looks fine

File: code/controllers/subsystems/initialization/codex_dump.dm [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	var/list/address_to_entry = list()
+	var/list/entry_to_address = list()
+	//var/list/seen_name_count =  list()
+	for(var/datum/codex_entry/codex_entry as anything in all_entries)
+		var/address = convert.convert_filename(codex_entry.get_dump_link_name())
+		address_to_entry[address] = codex_entry
+		entry_to_address[codex_entry] = address
+		address_to_body[address] = convert.convert_body(codex_entry.name, codex_entry.get_codex_body(include_header = FALSE, include_footer = FALSE))
+
+	// Copied from del-the-world UT exceptions list.
```
Comment (out-of-phaze) [inline code/controllers/subsystems/initialization/codex_dump.dm]: not quite. you do all subtypes of `/atom/movable`, while dtw only instantiates `typesof(/obj/item, /obj/effect, /obj/structure, /obj/machinery, /obj/vehicle, /mob) - ignore`. for example, `/obj/random` is excluded implicitly from dtw (maybe uh, bit of an oversight there actually?). excluding it here would fix the runtime, i've got no clue why missiles are failing GC though

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**